common/keyboard.go

```diff
@@ -253,6 +251,36 @@ func (k *Keyboard) modifierBitFromKeyName(key string) int64 {
 	return 0
 }
 
+func (k *Keyboard) comboPress(keys string, opts *KeyboardOptions) error {
+	if opts.Delay != 0 {
+		t := time.NewTimer(time.Duration(opts.Delay) * time.Millisecond)
+		select {
+		case <-k.ctx.Done():
+			t.Stop()
+		case <-t.C:
+		}
```

Comment:
This timer will leak if `t.C` matches.

There are similar leaks in the code (I guess we missed them before). Maybe it's time to fix them all.

Reply:
The timer creates a goroutine, but `t.Stop` is only called to stop the goroutine in the first `case`.

Reply:
Ok, as far as i can tell, the timer doesn't need to be stopped since we've read off of the channel. Although we should be reading off of the channel when we call `Stop` in the case above. Sound correct to you? Source: https://pkg.go.dev/time#NewTimer

Reply:
You're right, sorry 🤦

Reply:
Ah, another idea. Why don't we just create a child timeout context? And block there?

```go
if opts.Delay != 0 {
	ctx, _ := context.WithTimeout(k.ctx, time.Duration(opts.Delay) * time.Millisecond)
	<-ctx.Done()
}
```